orchestration/dagster_orchestration/hca_orchestration/tests/support/filesystem.py

```diff
@@ -0,0 +1,21 @@
+from contextlib import contextmanager
+import os
+import shutil
+from typing import Iterator
+
+
+# works as a temporary directory, but with a fixed name
+@contextmanager
+def EphemeralNamedDirectory(dirname: str, parent_directory: str) -> Iterator[str]:
```

Comment:
It's surprising that python does not provide this out of the box with TemporaryDirectory.

Reply:
I suspect it's something to do with being unable to make a temporary directory securely (which is one of the guarantees the `tempfile` module makes) if it's made with a fixed name.